src/fiatconnect/kyc/KycExpired.tsx

```diff
@@ -1,26 +1,121 @@
-import * as React from 'react'
+import React, { useMemo } from 'react'
 import { StyleSheet, Text } from 'react-native'
 import { SafeAreaView } from 'react-native-safe-area-context'
 import { StackScreenProps } from '@react-navigation/stack'
 import { StackParamList } from 'src/navigator/types'
 import { Screens } from 'src/navigator/Screens'
+import { navigate } from 'src/navigator/NavigationService'
+import { useTranslation } from 'react-i18next'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { FiatExchangeEvents } from 'src/analytics/Events'
+import { KycStatus as FiatConnectKycStatus } from '@fiatconnect/fiatconnect-types'
+import { convertCurrencyToLocalAmount } from 'src/localCurrency/convert'
+import fontStyles from 'src/styles/fonts'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+
+import { useSelector } from 'react-redux'
+import { localCurrencyExchangeRatesSelector } from 'src/localCurrency/selectors'
+import getNavigationOptions from 'src/fiatconnect/kyc/getNavigationOptions'
 
 type Props = StackScreenProps<StackParamList, Screens.KycExpired>
 
-// TODO implement designs:
-// https://www.figma.com/file/bwIY7peeykI8K4wTrXs4iN/Fiat-Connect?node-id=2945%3A41941
-function KycExpired(_props: Props) {
+function KycExpired({ route, navigation }: Props) {
+  const { quote, flow } = route.params
+
+  const exchangeRates = useSelector(localCurrencyExchangeRatesSelector)
+  useMemo(() => {
+    navigation.setOptions(
+      getNavigationOptions({
+        fiatConnectKycStatus: FiatConnectKycStatus.KycExpired,
+        quote,
+        exchangeRates,
+      })
+    )
+  }, [exchangeRates])
+
+  const { t } = useTranslation()
+
+  const onPressTryAgain = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.cico_fc_kyc_status_try_again, {
+      provider: quote.getProviderId(),
+      flow,
+      fiatConnectKycStatus: FiatConnectKycStatus.KycExpired,
+    })
+    navigate(Screens.KycLanding, {
+      quote,
+      flow,
+      step: 'one',
+    })
+  }
+  const onPressSwitch = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.cico_fc_kyc_status_switch_method, {
+      provider: quote.getProviderId(),
+      flow,
+      fiatConnectKycStatus: FiatConnectKycStatus.KycExpired,
+    })
+    navigate(Screens.SelectProvider, {
+      flow,
+      selectedCrypto: quote.getCryptoType(),
+      amount: {
+        crypto: Number(quote.getCryptoAmount()),
+        fiat: Number(
+          convertCurrencyToLocalAmount(
+            quote.getCryptoAmount(),
+            exchangeRates[quote.getCryptoType()]
+          )
```

Comment:
for cash-ins this could create some weird edge cases where a user tries to cash in with some amount, say $100, then they get to KycExpired, then when they navigate back to SelectProvider the amount changes to $95. because the currency is converted by the provider to some crypto amount, and then by us back to a fiat amount.

in short, I think just using quote.getFiatAmount() here is better

Reply:
Sure, makes sense -- that actually simplifies things a ton.